util/update-repo.sh

```diff
@@ -0,0 +1,42 @@
+#!/bin/bash
+
+if test "$#" -ne 1; then
+    echo "$0 software-stack | data | compute | io | app-interact" 1>&2
+    exit 1
+fi
+
+git stash
```

Comment:
This is an uncanny solution, as students will have the impression they have lost their previous work. Furthermore, popping the stash will likely result in merge conflicts as well. The best solution here IMO is for students to simply commit their changes locally and then safely switch branches.